### PR TITLE
Charms-ha update for 2.0

### PR DESCRIPTION
**Pull request by degville**
#1021 

- brief rewrite to clarify how HA works with charms, as well as what availability zones and distribution groups do
- added clouds that support availability zones
- fixed reference to state server (more to come)

### CODE REVIEW

File: src/en/charms-ha.md [hunk 1]
```diff
@@ -1,71 +1,77 @@
...
 
 ```bash
 juju deploy -n 10 <application>
 ```
-
-The way this works depends on whether Juju uses Availability Zones or
-Availability Sets for that provider.
-
+The way this works depends on whether Juju uses 'availability zones' or
+'availability sets' for that provider.
```
Comment (evilnick) [inline src/en/charms-ha.md]: the word 'provider' is outlawed!

File: src/en/charms-ha.md [hunk 2]
```diff
@@ -1,71 +1,77 @@
...
-provider-specific settings.
+Availability zones allow for the automatic and uniform distribution of units
+across a region. A new instance, for example, will be allocated the zone
+with the fewest members of its distribution group.
+
+Juju supports such zones on Google Compute Engine, VMware vSphere, Amazon's
+EC2, OpenStack-based clouds (Havana or newer) and
+[MAAS](http://maas.io/docs/installconfig-zones). See the
+[Clouds](./clouds.html) section of the documentation for more details on these
+and other provider-specific settings.
```
Comment (evilnick) [inline src/en/charms-ha.md]: cloud-specific

File: src/en/charms-ha.md [hunk 3]
```diff
@@ -1,71 +1,77 @@
...
 overall.  Exposed ports are automatically load-balanced across all units within
 the application.  Using availability sets disables manual placement and the
 "add-machine" command.
 
-In Juju 1.20 and later, new Azure environments use availability sets by
-default. To disable availability sets, the 'availability-sets-enabled' option
-must be set in environments.yaml like so:
+New Azure environments use availability sets by default. To disable
+this behaviour, the 'availability-sets-enabled' option must be set in
+environments.yaml like so:
```
Comment (evilnick) [inline src/en/charms-ha.md]: there is no environments.yaml. I guess this needs to be set using set-model-config

File: src/en/charms-ha.md [hunk 4]
```diff
@@ -1,71 +1,77 @@
...
 
 ```yaml
   availability-sets-enabled: false
 ```
+!!! Note: By disabling availability sets , you will lose Azure's SLA
```
Comment (evilnick) [inline src/en/charms-ha.md]: delete extra space before comma

File: src/en/charms-ha.md [hunk 5]
```diff
@@ -1,71 +1,77 @@
...
 units are deployed, Azure guarantees 99.95% availability of the application
 overall.  Exposed ports are automatically load-balanced across all units within
 the application.  Using availability sets disables manual placement and the
 "add-machine" command.
 
-In Juju 1.20 and later, new Azure environments use availability sets by
-default. To disable availability sets, the 'availability-sets-enabled' option
-must be set in environments.yaml like so:
+New Azure environments use availability sets by default. To disable
+this behaviour, the 'availability-sets-enabled' option must be set in
```
Comment (evilnick) [inline src/en/charms-ha.md]: this also confused me the first time i read it. perhaps : 'the availability-sets-enabled' option must be set to 'false' like this:

or something

File: src/en/controllers-ha.md [hunk 6]
```diff
@@ -1,13 +1,13 @@
...
 # High availability
 
-Juju high availability (HA) implies the presence of 3 or more (up to 7)
-controllers (per cloud) one of which is the *master*. Automatic failover occurs
-should the master lose connectivity.
+To ensure the high availability of deployed applications, the Juju controller
+must itself be highly available. This is accomplished by activating 'highly
+available' mode which ensures 3 or more (up to 7) controllers (per cloud) are
+available, one of which is the *master*.  Automatic failover occurs should the
+master lose connectivity.
```
Comment (evilnick) [inline src/en/controllers-ha.md]: redo as discussed

File: src/en/charms-ha.md [hunk 7]
```diff
@@ -56,19 +56,18 @@ overall.  Exposed ports are automatically load-balanced across all units within
...
 "add-machine" command.
 
-New Azure environments use availability sets by default. To disable
-this behaviour, the 'availability-sets-enabled' option must be set in
-environments.yaml like so:
+New Azure environments use availability sets by default. This behaviour can be
+disabled only when bootstrapping the cloud by adding
+'availability-sets-enabled=false' as a configuration option:
 
 ```yaml
```
Comment (matthewhelmke) [inline src/en/charms-ha.md]: Did you want to continue to format this as 'yaml' or instead change it to 'bash', like most other text boxes?

**Pull request merged**